index.bs

```diff
@@ -1407,13 +1384,75 @@ session.CapabilityRequest = {
     ? socksProxy: text,
     ? socksVersion: 0..255,
   },
+  ? webSocketUrl: bool,
   Extensible
 };
 </pre>
 
 The <code>session.CapabilityRequest</code> type represents a specific set of
 requested capabilities.
 
+WebDriver BiDi defines [=additional WebDriver capability|additional WebDriver
+capabilities=]. The following tables enumerates the capabilities each
+implementation must support for WebDriver BiDi.
+
+<pre class=simpledef>
+Capability: <dfn export>WebSocket URL</dfn>
+Key: "<code>webSocketUrl</code>"
+Value type: boolean
+Description: Defines the current session's support for bidirection connection.
+</pre>
+
+<pre class=simpledef>
+Capability: <dfn export>Drag and Drop Support</dfn>
+Key: "<code>enableDragAndDrop</code>"
+Value type: boolean
+Description: Defines the current session's drag and drop support.
+</pre>
+
+<div algorithm="webSocketUrl capability deserialization algorithm">
+The [=additional capability deserialization algorithm=] for the
+"<code>webSocketUrl</code>" capability, with parameter |value| is:
+
+1. If |value| is not a boolean, return [=error=] with [=error code|code=]
+   [=invalid argument=].
+
+1. Return [=success=] with data |value|.
+
+</div>
+
+<div algorithm="webSocketUrl capability serialization algorithm">
+The [=matched capability serialization algorithm=] for the "<code>webSocketUrl</code>" capability,
+with parameter |value| is:
+
+1. If |value| is false, return [=success=] with data null.
+
+1. Return [=success=] with data true.
+
+</div>
+
+<div algorithm="enableDragAndDrop capability deserialization algorithm">
+The [=additional capability deserialization algorithm=] for the
+"<code>enableDragAndDrop</code>" capability, with parameter |value| is:
+
+1. If |value| is not a boolean, return [=error=] with [=error code|code=]
+   [=invalid argument=].
+
+1. Return [=success=] with data |value|.
+
+</div>
+
+<div algorithm="enableDragAndDrop capability serialization algorithm">
+The [=matched capability serialization algorithm=] for the "<code>enableDragAndDrop</code>" capability,
+with parameter |value| is:
+
+1. If |value| is false, the [=endpoint node=] must disable [=drag
```

Comment:
I think this is too broad: we don't want to disable drag and drop for other sessions and/or regular user input. I believe we only want to make sure that the actions dispatched by the WebDriver BiDi client don't get converted into a drag and drop, right?

Reply:
Sadly the problem is drag and drop must be completely disabled on the endpoint. This is because while drag and drop is happening, _all_ inputs (other sessions, regular user input, etc) gets ignored.

Reply:
Also, it's impossible to separate regular user input and BiDi input for drag and drop. The reason is that drag and drop is predominantly a OS API, meaning when it happens, you must either delegate it to the OS or delegate it to the browser (similar to how Chromium does it). You cannot have both control it at the same time since the OS expects to be able to end the drag itself.